src/services/challenges_repository.js

```diff
@@ -45,4 +46,12 @@ export default class ChallengesRepository {
     }
     return this.challengesWrapper.resolve(challengeId);
   }
+
+  async wasChallengeRejectedRecently(challengeId) {
+    return (await this.db.collection('rejectedChallenges').findOne({challengeId})) !== null;
+  }
+
+  async markChallengeAsRejected(challengeId) {
+    await this.db.collection('rejectedChallenges').insertOne({challengeId, createdAt: Date.now()});
```

Comment:
`createdAt` sounds like timestamp of challenge creation. `rejectedAt`? `rejectionTimestamp`?

Reply:
It's a mongo document creation time. Seems fine to me